solr/core/src/java/org/apache/solr/cloud/ElectionContext.java

```diff
@@ -580,7 +580,7 @@ public void publishActiveIfRegisteredAndNotActive(SolrCore core) throws Exceptio
         ClusterState clusterState = zkStateReader.getClusterState();
         Replica rep = getReplica(clusterState, collection, leaderProps.getStr(ZkStateReader.CORE_NODE_NAME_PROP));
         if (rep == null) return;
-        if (rep.getState() != Replica.State.ACTIVE || core.getCoreDescriptor().getCloudDescriptor().getLastPublished() != Replica.State.ACTIVE) {
+        if (zkStateReader.getShardStateProvider(collection).getState(rep) != Replica.State.ACTIVE || core.getCoreDescriptor().getCloudDescriptor().getLastPublished() != Replica.State.ACTIVE) {
```

Comment:
Should this be `! isActive(rep)` instead of `getState(rep) != ACTIVE`? There's a lot of places where we still call `getState`.